lib/base/threadpool.hpp

```diff
@@ -36,7 +37,7 @@ class ThreadPool
 public:
 	typedef std::function<void ()> WorkFunction;
 
-	ThreadPool(size_t threads = std::thread::hardware_concurrency() * 2u);
+	ThreadPool(size_t threads = Configuration::Concurrency * 2u);
```

Comment:
The main use of this seems to be the `ThreadPool` in `Application::GetTP()` and it looks like this one is initialized too early for this change to make a difference.

Trying to set `Concurrency` from the config files doesn't work at all:
```
[2023-01-27 15:45:28 +0100] critical/config: Error: Configuration attribute 'Concurrency' is read-only.
```

So it looks like `-DConcurrency=42` is the only option. But even this is too late, output with extra logging in `Configuration::SetConcurrency` and `ThreadPool::ThreadPool`:

```
ctor ThreadPool: 16
ctor ThreadPool: 16
SetConcurrency: 42
[2023-01-27 15:48:19 +0100] information/cli: Icinga application loader (version: v2.13.0-578-gd10e7ed68; debug)
```